### ISSUE TITLE
TabFocusNavigation="Once" doesn't always work right with SHIFT+TAB

### ISSUE THREAD

**Issue by randy-flynn** [microsoft/microsoft-ui-xaml#1363]
**Describe the bug**

When you create a Panel with TabFocusNavigation set to "Once" and then have an additional level of hierarchy between that Panel and the focusable children the SHIFT+TAB behavior is not correct. See attached image for code.

![image](https://user-images.githubusercontent.com/48224713/65291484-6f988a80-db08-11e9-999d-90342d551ca1.png)

**Steps to reproduce the bug**
1. Create project with code in attached image. Build & run.
2. TAB into the StackPanel - note focus correctly goes to ONE.
3. TAB out of the StackPanel.
4. SHIFT+TAB back into the StackPanel - note focus correctly goes to THREE.
5. SHIFT+TAB again - note that focus incorrectly goes to TWO.

**Expected behavior**
The focus should have exited to the top of the StackPanel.

**Comment by Austin-Lamb** [comment]
Thanks for reporting this - I can confirm I see the same problem, so your minimal repro is great.  I'm going to put this on our backlog for the moment, as we can only fix this in WinUI 3 due to it being in the core framework.

**Comment by rectified95** [comment]
@Austin-Lamb Has the team made any progress on fixing this? I can confirm this is a high-priority bug that we're seeing in RNW as well.

**Comment by lyahdav** [comment]
Here's a workaround in WinRT/C++, but should work if ported to C# as well. Just set the `KeyDown` and `GettingFocus` handlers below to the element that you're setting TabFocusNavigation=Once. In our case we also set XYFocusKeyboardNavigation=Enabled, so some of that code might not be relevant. Also `KeyboardHelper::IsModifiedKeyPressed` below is defined [here](https://github.com/microsoft/react-native-windows/blob/003602b90aeda1b2a43702c6a9307c0f01010e59/vnext/Microsoft.ReactNative/Views/KeyboardEventHandler.cpp#L624).
```
bool IsElementChildOf(winrt::DependencyObject element, winrt::DependencyObject expectedParent) {
  winrt::DependencyObject parent = element;
  do {
    parent = winrt::VisualTreeHelper::GetParent(parent);
    if (parent == expectedParent) {
      return true;
    }
  } while (parent != nullptr);
  return false;
}

bool IsElementWithinAnotherFocusableElementInContainer(winrt::DependencyObject element, winrt::FrameworkElement container) {
  auto parent = element;
  do {
    parent = winrt::VisualTreeHelper::GetParent(parent);
    auto control = parent.try_as<winrt::Control>();
    if (control != nullptr && control.IsTabStop() && control.IsEnabled()) {
      return true;
    }
  } while (parent != nullptr && parent != container);
  return false;
}

winrt::ScrollViewer GetFirstChildScrollViewer(winrt::DependencyObject element) {
  if (auto scrollViewer = element.try_as<winrt::ScrollViewer>()) {
    return scrollViewer;
  }

  auto childrenCount = winrt::VisualTreeHelper::GetChildrenCount(element);
  for (auto i = 0; i < childrenCount; i++) {
    auto child = winrt::VisualTreeHelper::GetChild(element, i);
    if (auto scrollViewer = GetFirstChildScrollViewer(child)) {
      return scrollViewer;
    }
  }

  return nullptr;
}

void KeyDown(
    const winrt::IInspectable& sender,
    const winrt::KeyRoutedEventArgs& e) {
  if (e.Key() == winrt::Windows::System::VirtualKey::Tab) {
    auto senderFrameworkElement = sender.try_as<xaml::FrameworkElement>();
    auto xamlRoot = senderFrameworkElement.XamlRoot();

    // With XYFocusKeyboardNavigation enabled, the way to focus a child of a
    // focusable element is with tab. Then shift+tab focuses the parent. If
    // we're in that case then let Xaml handle the shift+tab.
    auto focusedElement = winrt::FocusManager::GetFocusedElement(xamlRoot)
                              .try_as<winrt::DependencyObject>();
    if (IsElementWithinAnotherFocusableElementInContainer(
            focusedElement, senderFrameworkElement)) {
      return;
    }

    auto const& coreWindow = winrt::CoreWindow::GetForCurrentThread();
    auto isShiftDown = KeyboardHelper::IsModifiedKeyPressed(
        coreWindow, winrt::Windows::System::VirtualKey::Shift);

    if (isShiftDown) {
      if (auto scrollViewer = GetFirstChildScrollViewer(senderFrameworkElement)) {
        // Since we leave the element by moving focus backwards
        // repeatedly, we don't want any ScrollViewer to scroll during
        // this process.
        scrollViewer.BringIntoViewOnFocusChange(false);
      }
      winrt::FindNextElementOptions findNextElementOptions =
          winrt::FindNextElementOptions();
      findNextElementOptions.SearchRoot(xamlRoot.Content());

      auto didMoveFocus = false;
      do {
        didMoveFocus = winrt::FocusManager::TryMoveFocus(
            winrt::FocusNavigationDirection::Previous, findNextElementOptions);
        focusedElement = winrt::FocusManager::GetFocusedElement(xamlRoot)
                             .try_as<winrt::DependencyObject>();
      } while (didMoveFocus &&
               IsElementChildOf(focusedElement, senderFrameworkElement));
      e.Handled(true);
    }
  }
}

void GettingFocus(
    const winrt::IInspectable& sender,
    const winrt::GettingFocusEventArgs& e) {
  auto senderDependencyObject = sender.as<winrt::DependencyObject>();
  auto isOldFocusedElementOutside = e.OldFocusedElement() != nullptr && !IsElementChildOf(e.OldFocusedElement(), senderDependencyObject);

  if (isOldFocusedElementOutside) {
    if (auto scrollViewer =
            GetFirstChildScrollViewer(senderDependencyObject)) {
      // Revert this property from above.
      scrollViewer.BringIntoViewOnFocusChange(true);
    }
  }
}
```

**Comment by JesseCol** [comment]
This is fixed in the 1.4 release.

**Comment by duncanmacmichael** [comment]
Fixed in 1.4.0. @bpulliam please close this issue as I cannot. :)
